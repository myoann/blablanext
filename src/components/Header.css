.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 30px 128px;
  border-bottom: 2px solid #aaaaaa;
}

.header a {
  display: flex;
  align-items: center;
  gap: 30px;
  text-decoration: none;
}

.logoTitle {
  font-size: 40px;
  font-weight: 900;
  color: #008ec2;
  font-style: italic;
  text-transform: uppercase;
}
