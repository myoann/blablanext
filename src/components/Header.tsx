import Image from "next/image";
import Link from "next/link";

import "./Header.css";

const Header = () => (
  <header className="header">
    <Link href="/">
      <span className="logoTitle">Carpool</span>
    </Link>

    <Image src="/logo.png" alt="Yoann Moise Logo" width={50} height={50} />

  </header>
);

export default Header;
