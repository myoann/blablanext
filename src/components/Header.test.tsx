import { render, screen } from "@testing-library/react";

import Header from "./Header";

describe("Header", () => {
  it("matches the snapshot", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });

  it("has a a link which redirects to /", () => {
    render(<Header />);

    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "/");
  });

  it("has a text with the name of the website (CARPOOL)", () => {
    render(<Header />);

    const logoTitle = screen.getByText("Carpool");
    expect(logoTitle).toBeInTheDocument();
  });
});
