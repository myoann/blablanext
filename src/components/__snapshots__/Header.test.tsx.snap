// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <a
      href="/"
    >
      <span
        class="logoTitle"
      >
        Carpool
      </span>
    </a>
    <img
      alt="Yoann Moise Logo"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=%2Flogo.png&w=128&q=75"
      srcset="/_next/image?url=%2Flogo.png&w=64&q=75 1x, /_next/image?url=%2Flogo.png&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
  </header>
</div>
`;
